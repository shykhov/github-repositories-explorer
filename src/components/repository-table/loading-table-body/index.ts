export * from './loading-table-body';
