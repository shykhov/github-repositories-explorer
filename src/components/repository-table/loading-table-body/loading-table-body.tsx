import React, { FC } from 'react';
import ContentLoader from 'react-content-loader';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import range from 'lodash/range';

const mockData = range(10);

export const LoadingTableBody: FC = () => (
  <>
    {mockData.map((value: number) => (
      <TableRow key={value} role="checkbox">
        <TableCell colSpan={3} padding="none">
          <ContentLoader foregroundColor="#BCCCDC" backgroundColor="#F0F4F8" height={49} width={1024} speed={2}>
            <rect x="0" y="0" rx="0" ry="0" width="1024" height="49" />
          </ContentLoader>
        </TableCell>
      </TableRow>
    ))}
  </>
);
