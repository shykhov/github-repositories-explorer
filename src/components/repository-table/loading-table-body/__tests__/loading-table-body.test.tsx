import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';

import { LoadingTableBody } from '../loading-table-body';

describe('<LoadingTableBody />', () => {
  let renderedComponent: ShallowWrapper;

  beforeEach(() => {
    renderedComponent = shallow(<LoadingTableBody />);
  });

  it('should render LoadingTableBody', () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});
