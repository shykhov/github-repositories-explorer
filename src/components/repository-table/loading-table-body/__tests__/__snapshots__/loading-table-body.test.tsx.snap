// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingTableBody /> should render LoadingTableBody 1`] = `
<Fragment>
  <WithStyles(ForwardRef(TableRow))
    key="0"
    role="checkbox"
  >
    <WithStyles(ForwardRef(TableCell))
      colSpan={3}
      padding="none"
    >
      <ContentLoader
        backgroundColor="#F0F4F8"
        foregroundColor="#BCCCDC"
        height={49}
        speed={2}
        width={1024}
      >
        <rect
          height="49"
          rx="0"
          ry="0"
          width="1024"
          x="0"
          y="0"
        />
      </ContentLoader>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))
    key="1"
    role="checkbox"
  >
    <WithStyles(ForwardRef(TableCell))
      colSpan={3}
      padding="none"
    >
      <ContentLoader
        backgroundColor="#F0F4F8"
        foregroundColor="#BCCCDC"
        height={49}
        speed={2}
        width={1024}
      >
        <rect
          height="49"
          rx="0"
          ry="0"
          width="1024"
          x="0"
          y="0"
        />
      </ContentLoader>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))
    key="2"
    role="checkbox"
  >
    <WithStyles(ForwardRef(TableCell))
      colSpan={3}
      padding="none"
    >
      <ContentLoader
        backgroundColor="#F0F4F8"
        foregroundColor="#BCCCDC"
        height={49}
        speed={2}
        width={1024}
      >
        <rect
          height="49"
          rx="0"
          ry="0"
          width="1024"
          x="0"
          y="0"
        />
      </ContentLoader>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))
    key="3"
    role="checkbox"
  >
    <WithStyles(ForwardRef(TableCell))
      colSpan={3}
      padding="none"
    >
      <ContentLoader
        backgroundColor="#F0F4F8"
        foregroundColor="#BCCCDC"
        height={49}
        speed={2}
        width={1024}
      >
        <rect
          height="49"
          rx="0"
          ry="0"
          width="1024"
          x="0"
          y="0"
        />
      </ContentLoader>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))
    key="4"
    role="checkbox"
  >
    <WithStyles(ForwardRef(TableCell))
      colSpan={3}
      padding="none"
    >
      <ContentLoader
        backgroundColor="#F0F4F8"
        foregroundColor="#BCCCDC"
        height={49}
        speed={2}
        width={1024}
      >
        <rect
          height="49"
          rx="0"
          ry="0"
          width="1024"
          x="0"
          y="0"
        />
      </ContentLoader>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))
    key="5"
    role="checkbox"
  >
    <WithStyles(ForwardRef(TableCell))
      colSpan={3}
      padding="none"
    >
      <ContentLoader
        backgroundColor="#F0F4F8"
        foregroundColor="#BCCCDC"
        height={49}
        speed={2}
        width={1024}
      >
        <rect
          height="49"
          rx="0"
          ry="0"
          width="1024"
          x="0"
          y="0"
        />
      </ContentLoader>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))
    key="6"
    role="checkbox"
  >
    <WithStyles(ForwardRef(TableCell))
      colSpan={3}
      padding="none"
    >
      <ContentLoader
        backgroundColor="#F0F4F8"
        foregroundColor="#BCCCDC"
        height={49}
        speed={2}
        width={1024}
      >
        <rect
          height="49"
          rx="0"
          ry="0"
          width="1024"
          x="0"
          y="0"
        />
      </ContentLoader>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))
    key="7"
    role="checkbox"
  >
    <WithStyles(ForwardRef(TableCell))
      colSpan={3}
      padding="none"
    >
      <ContentLoader
        backgroundColor="#F0F4F8"
        foregroundColor="#BCCCDC"
        height={49}
        speed={2}
        width={1024}
      >
        <rect
          height="49"
          rx="0"
          ry="0"
          width="1024"
          x="0"
          y="0"
        />
      </ContentLoader>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))
    key="8"
    role="checkbox"
  >
    <WithStyles(ForwardRef(TableCell))
      colSpan={3}
      padding="none"
    >
      <ContentLoader
        backgroundColor="#F0F4F8"
        foregroundColor="#BCCCDC"
        height={49}
        speed={2}
        width={1024}
      >
        <rect
          height="49"
          rx="0"
          ry="0"
          width="1024"
          x="0"
          y="0"
        />
      </ContentLoader>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))
    key="9"
    role="checkbox"
  >
    <WithStyles(ForwardRef(TableCell))
      colSpan={3}
      padding="none"
    >
      <ContentLoader
        backgroundColor="#F0F4F8"
        foregroundColor="#BCCCDC"
        height={49}
        speed={2}
        width={1024}
      >
        <rect
          height="49"
          rx="0"
          ry="0"
          width="1024"
          x="0"
          y="0"
        />
      </ContentLoader>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
</Fragment>
`;
